### PR TITLE
MQTT Integration

### PR DESCRIPTION
**Pull request by paynemiller92**
### Overview of Changes
--
1. Removed GCM dependecy and references.
2. Installed Mqtt Client dependency
3. Configured MQTT client to point to server

### Notes
--
The API team will specify the actual topic I need to subscribe to. Currently, I am subscribing to the topic `myTopic` as a placeholder. Addresses issue https://github.com/PushFish/pushfish-android/issues/1.

### Sample Log

```java
10-23 19:04:10.147 27591-27635/io.Pushjet.api I/io.Pushjet.api.PushListActivity: MQTT Connection successful!
```

### CODE REVIEW

**Comment by VoliKoN** [comment]
You should subscribe to the topic with the name of your uuid

**Comment by paynemiller92** [comment]
@VoliKoN the "name" of the uuid? The uuid is usually a GUID, no?

File: app/build.gradle [hunk 1]
```diff
@@ -39,7 +39,9 @@ dependencies {
     compile 'com.android.support:appcompat-v7:24.0.0-alpha2'
     compile 'com.google.android.gms:play-services-ads:8.4.0'
     compile 'com.google.android.gms:play-services-auth:8.4.0'
```
Comment (VoliKoN) [inline app/build.gradle:41]: Shouldn't we remove all mentions of play services?

File: app/src/main/java/io/Pushjet/api/PushListActivity.java [hunk 2]
```diff
@@ -148,17 +159,72 @@ public boolean onItemLongClick(AdapterView<?> parent, View v, int position, long
...
+            subscribeToMqttTopic();
+            mqttClient.connect(constructConnectionOtions());
+        } catch (MqttException e) {
+            e.printStackTrace();
         }
     }
 
+    private void configureMqttClient() {
+        try {
+            mqttClient = new MqttAsyncClient("tcp://test-api.push.fish:1883", "android", new MemoryPersistence());
```
Comment (VoliKoN) [inline app/src/main/java/io/Pushjet/api/PushListActivity.java:174]: This should be configurable, with the default the same as the api URL with 1883 port

**Comment by OliverParoczai** [comment]
The correct UUID shows up in the settings page of the app, i think you can use the code that is used there for showing the UUID

**Comment by OliverParoczai** [comment]
This is the UUID the app uses: https://github.com/PushFish/pushfish-android/blob/master/app/src/main/java/io/Pushjet/api/SettingsActivity.java#L182

File: app/build.gradle (hunk 1 above)
Comment (paynemiller92) [reply app/build.gradle:41]: No. Google Auth and Google Ads are most likely being used in other places. GMS was the only Google Play component specified to be removed. If we no longer have a need for those components, only then should we remove them.

File: app/src/main/java/io/Pushjet/api/PushListActivity.java (hunk 2 above)
Comment (paynemiller92) [reply app/src/main/java/io/Pushjet/api/PushListActivity.java:174]: We can make them a build variable.

Comment (paynemiller92) [reply app/src/main/java/io/Pushjet/api/PushListActivity.java:174]: Do you have other potential configurations I may use as examples?

Comment (VoliKoN) [reply app/src/main/java/io/Pushjet/api/PushListActivity.java:174]: If the server url is configurable on the app's settings, this should also be.

File: app/build.gradle (hunk 1 above)
Comment (VoliKoN) [reply app/build.gradle:41]: I don't think a foss app should have either of those things.
In addition it's needed for uploading to f-droid

Comment (paynemiller92) [reply app/build.gradle:41]: @VoliKoN @Olika120 @Leopere are we using the Google Play Components Ads and Auth anywhere else in the app? If not, can we open a separate issue to remove those?

Comment (VoliKoN) [reply app/build.gradle:41]: #4 already written

Comment (paynemiller92) [reply app/build.gradle:41]: Perfect!

**Pull request merged**